File: Dart/q5.dart
import 'dart:io';

int gcd(int a, int b) {
  while (b != 0) {
    int temp = b;
    b = a % b;
    a = temp;
  }
  return a;
}

int Lcm(int a, int b) {
  return (a * b) ~/ gcd(a, b);
}

int smallest_multiple(int start, int end) {
  int result = start;
  for (int i = start + 1; i <= end; i++) {
    result = Lcm(result, i);
  }
  return result;
}

void main() {
  print("Enter the range to find Smallest Multiple");
  print("From Range");
  int from_Range = int.parse(stdin.readLineSync()!);
  print("To range");
  int to_range = int.parse(stdin.readLineSync()!);
  int result = smallest_multiple(from_Range, to_range);
  print(result);
}

File: Python/q5.py
def gcd(a, b):
    while b != 0:
        temp = b
        b = a % b
        a = temp
    return a

def lcm(a, b):
    return (a * b) // gcd(a, b)

def smallest_multiple(start, end):
    result = start
    for i in range(start + 1, end):
        result = lcm(result, i)
    return result 

from_start = int(input("Enter the start range: "))
form_end = int(input("Enter the end range: "))

final_result = smallest_multiple(from_start, form_end)
print(final_result)